lib/log.h

```diff
@@ -71,16 +71,36 @@ extern enum metal_log_level metal_get_log_level(void);
 extern void metal_default_log_handler(enum metal_log_level level,
 				      const char *format, ...);
 
+/**
+ * @brief	used by the metal_log() macro to update the format string
+ * @fmt	format string passed from the metal_log() macro
+ *
+ * If ML_FUNC_LINE is defined this macro generates a unified format
+ * string for metal_log() and its convenience ml_*() macros, i.e. it
+ * adds function-name:line-number prefix to all log messages.
+ */
+#if defined(ML_FUNC_LINE)
+#define ml_fmt(fmt) "%s:%u " fmt, __func__, __LINE__
+#else	/* ML_FUNC_LINE */
+#define ml_fmt(fmt) fmt
+#endif	/* ML_FUNC_LINE */
+
 /**
  * Emit a log message if the log level permits.
  *
  * @param	level	Log level.
  * @param	...	Format string and arguments.
  */
-#define metal_log(level, ...)						       \
-	((level <= _metal.common.log_level && _metal.common.log_handler) \
-	       ? (void)_metal.common.log_handler(level, __VA_ARGS__)	       \
-	       : (void)0)
+#define metal_log(level, fmt, args ...)					\
+do {									\
+	if (_metal.common.log_handler && level <= _metal.common.log_level) \
+		_metal.common.log_handler(level, ml_fmt(fmt), ##args);	\
+} while (0)
+
+#define ml_err(fmt, args ...) metal_log(METAL_LOG_ERROR, fmt, ##args)
+#define ml_warn(fmt, args ...) metal_log(METAL_LOG_WARNING, fmt, ##args)
+#define ml_info(fmt, args ...) metal_log(METAL_LOG_INFO, fmt, ##args)
+#define ml_dbg(fmt, args ...) metal_log(METAL_LOG_DEBUG, fmt, ##args)
```

Comment:
These macros seem useless

Reply:
We have discussed these in our RP call and agreed to add "convenience macros ml_err, ml_warn, ml_info, ml_dbg to avoid using excessively long and redundant "metal_log(METAL_LOG_*".

Reply:
@arnopo I see values in these macros. Developers can use short macros with clean interface for logging. Instead of metal_log(METAL_LOG_* it is cleaner to use ml_* macros which aligns kernel's pr_* macros. So, we can keep macros and new logs can be added with these macros.

Having said that, we can replace ml_dbg with ml_debug.

Reply:
That's valid. 
If we keep them, we have to find a naming started by "metal_" to be consistent.
what about metal_err, metal_warb, metal_info, metal_debug (or metal_dbg)?

Reply:
If changing form ml_err to metal_err, etc. is a condition of acceptance, I'd like to use metal_dbg. We should not impose higher typing, space cost if we want to encourage more messages for error reporting, debugging, etc. Here is the length (cost) of each version:
```
26 metal_log(METAL_LOG_DEBUG,
12 metal_debug(
10 metal_dbg(
 7 ml_dbg(
```
I'd vote for the shorter version. In this context "ml" stands for metal_log or metal and I can comment in the log.h.

Reply:
I am okay with metal_dbg as well. Please do the change accordingly.

Reply:
Thank you. Done.